types/index.d.ts

```diff
@@ -511,16 +512,19 @@ export class Transaction<
   addSignature(publicKey: string, signature: string): void;
   getKeypairSignature(keypair: Keypair): string;
   hash(): Buffer;
+  isFeeBump(): boolean;
   sign(...keypairs: Keypair[]): void;
   signatureBase(): Buffer;
   signHashX(preimage: Buffer | string): void;
   toEnvelope(): xdr.TransactionEnvelope;
   toXDR(): string;
+  static buildFeeBumpTransaction(feeSource: Keypair, fee: string, innerTx: xdr.TransactionEnvelope, networkPassphrase: string): Transaction;
 
   operations: TOps;
   sequence: string;
-  fee: number;
+  fee: number|string;
```

Comment:
seems weird to have the type be a number or a string. is the reason for keeping the number option for backwards compatibility?

Reply:
right, this is for backwards compatibility since we'll release this while protocol 12 is still running.

Reply:
it's possible to represent the protocol 12 fee as a string. however, users of the js sdk must be prepared for the new type. since this is a major version bump you could make the argument that it's ok to change the type from `number` to `string`. but since javascript is not statically typed I think users of the sdk will not realize the type change unless they look at the changelog

Reply:
right, this is focus only on TS users -- so having this will allow them to start preparing their code for both scenarios. I think eventually we can remove the typing once we are only returning `string`. 

This is similar to what we did to allow the transition from int to string on int64 values.

Reply:
@tamirmfs after thinking more about this, I think we don't need backwards compatibility for this field - we can change it to `string` and let it be a breaking change.

Reply:
I support that - getting this changed by downstream users sooner rather than later is a win